bootstrap-salt.sh

```diff
@@ -2430,10 +2404,10 @@ install_ubuntu_stable_deps() {
         # Versions starting with 2015.5.6, 2015.8.1 and 2016.3.0 are hosted at repo.saltstack.com
         if [ "$(echo "$STABLE_REV" | egrep '^(2015\.5|2015\.8|2016\.3|2016\.11|latest|archive\/)')" != "" ]; then
             # Workaround for latest non-LTS ubuntu
-            if [ "$DISTRO_MAJOR_VERSION" -eq 15 ]; then
+            if [ "$DISTRO_MAJOR_VERSION" -eq 16 ] && [ "$DISTRO_MINOR_VERSION" -eq 10 ]; then
                 echowarn "Non-LTS Ubuntu detected, but stable packages requested. Trying packages from latest LTS release. You may experience problems."
-                UBUNTU_VERSION=14.04
-                UBUNTU_CODENAME=trusty
+                UBUNTU_VERSION=16.04
+                UBUNTU_CODENAME=xenial
```

Comment:
@rallytime This is what I've mentioned in comments for PR #1001 .

The above and later code explicitly says:
**If** we are on 16.10 "yakkety" **then** go configure the SaltStack's repo for 16.04 "xenial", which is:
```
https://repo.saltstack.com/apt/ubuntu/16.04/amd64/latest/SALTSTACK-GPG-KEY.pub
```

This is the "best effort" approach introduced before for Ubuntu non-LTS releases and now it simply works for 16.10 too!
I have tested it and got latest Salt packages successfully installed (and running) from SaltStack's repo on "yakkety", while all other dependencies were fetched from Ubuntu's main and universe repositories.

I think this is really nice, now we have full Ubuntu 16.10 support! That's worth adding to the README :smiley: 

Anyhow, the changes in #1001 still make sense because for now the packages from "xenial" repo are compatible and installable on the "yakkety" system, but that could be changed any moment.